### PR TITLE
add user management role #2

### PR DESCRIPTION
**Pull request by IncredibleRichie**
#1083

### CODE REVIEW

File: tests/test_playbooks/users_role.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
+---
+# The following sensitive variables can be specified in the server.yml
+# ldap_host:
+# ldap_account:
+# ldap_password:
+# ldap_base_dn:
+# ldap_groups_base:
```
Comment (mdellweg) [inline tests/test_playbooks/users_role.yml]: Not true anymore?

Can you add information about the steps that are necessary to run / record this test?

File: roles/users/README.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+  - username: string **required**
+    firstname: string
+    lastname: string
+    email: string(email)
+    description: string
+    admin_rights: bool
+    user_password: string **required when internal**
+    default_location: string
+    default_organization: string
+    auth_source: sting **required**
```
Comment (mdellweg) [inline roles/users/README.md]:
```suggestion
    auth_source: string **required**
```

File: roles/users/README.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+validate_certs: bool defualt (true)
+```
+
+To switch on/off tasks use:
+
+``` yaml
+create_users: bool defualt (true)
+create_usergroups: bool defualt (true)
+enable_ldap_auth: bool defualt (true)
+map_external_usergroups: bool defualt (true)
```
Comment (mdellweg) [inline roles/users/README.md]:
```suggestion
create_users: bool default (true)
create_usergroups: bool default (true)
enable_ldap_auth: bool default (true)
map_external_usergroups: bool default (true)
```

Comment (mdellweg) [reply roles/users/README.md]: What is the usecase here? Would it make sense to just perform the actions if the corresponding data structures are not empty?

File: roles/users/README.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+#### Example values can be found in defaults/main.yaml
+
+Default:
+
+``` yaml
+server_url: string(url) **required**
+username: string **required**
+password: string
+password: string
+validate_certs: bool defualt (true)
```
Comment (mdellweg) [inline roles/users/README.md]: At least for this set of variables, there is a common text snippet used in the other roles, i think.

https://github.com/theforeman/foreman-ansible-modules/tree/develop/roles/manifest#role-variables

File: roles/users/README.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+```
+
+To switch on/off tasks use:
+
+``` yaml
+create_users: bool defualt (true)
```
Comment (mdellweg) [inline roles/users/README.md]: All top level vars need proper namespacing.
...even if it sounds silly.

File: tests/test_playbooks/users_role.yml (hunk 1 above)
Comment (IncredibleRichie) [reply tests/test_playbooks/users_role.yml]: I´ve added it to the server.example also. 
What do you mean exactly?

File: roles/users/README.md (hunk 2 above)
Comment (IncredibleRichie) [reply roles/users/README.md]: thx

File: roles/users/README.md (hunk 3 above)
Comment (IncredibleRichie) [reply roles/users/README.md]: I have a full playbook (vars) what describes my instance or in this case my user space, with the above vars in a playbook i can choose what to run without del/paste data.

File: tests/test_playbooks/users_role.yml (hunk 1 above)
Comment (mdellweg) [reply tests/test_playbooks/users_role.yml]: I think you mentioned you had a ldap container.
If that's something the developer needs to supply in order to record this test, it should be mentioned here.

Example:
https://github.com/theforeman/foreman-ansible-modules/blob/develop/tests/test_playbooks/realm.yml#L1

File: roles/users/README.md (hunk 3 above)
Comment (mdellweg) [reply roles/users/README.md]: I'm not sold on this. Do we have a similar thing in other roles?

Comment (mdellweg) [reply roles/users/README.md]: To be a bit more verbose: I thought, the deal of this role is to define your instance and the role will do every thing needed. I don't understand, why i should specify everything to then not use part of it. I could still be convinced though.

Maybe as a guideline: Those roles fall into the 80/20 rule. If you need something really special, you can write your own. We want to serve the common use case.

File: tests/test_playbooks/users_role.yml (hunk 1 above)
Comment (IncredibleRichie) [reply tests/test_playbooks/users_role.yml]: Ah ok added some information.

File: roles/users/README.md (hunk 4 above)
Comment (IncredibleRichie) [reply roles/users/README.md]: Applied

File: roles/users/README.md (hunk 5 above)
Comment (IncredibleRichie) [reply roles/users/README.md]: Ok, i´ve changed them.

File: roles/users/README.md (hunk 3 above)
Comment (IncredibleRichie) [reply roles/users/README.md]: You dont have to specify anything, only the data for enabled tasks.

I think this happens very often as an admin. Example: 
If an admin already has the playbook written and wants to add an internal user without going to the whole process with ldap etc. he can only activate the create user part and run it. 
On the other side you can only create a auth source and link groups without create/update plenty users.

**Pull request closed without merging**